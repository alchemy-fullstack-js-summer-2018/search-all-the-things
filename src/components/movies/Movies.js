import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Movie from './Movie';
import styles from './Movies.css';

export default class Movies extends Component {

    static propTypes = {
      movies: PropTypes.arrayOf(Object)
    };
  
    render() {
      const { movies } = this.props;
  
      return (
        <ul className={styles.movies}>
          {movies.map((movie, i) => (
            <Movie key={i} movie={movie}/>
          ))}
        </ul>
      );
    }
}