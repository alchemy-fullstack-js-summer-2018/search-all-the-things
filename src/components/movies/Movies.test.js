import * as React from 'react';
import { shallow, configure } from 'enzyme';
import Movies from './Movies';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const movies = [{}, {}, {}];

describe('Movies test', () => {
    
  it('Movies renders as designed', () => {
    const wrapper = shallow(<Movies movies={movies}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});