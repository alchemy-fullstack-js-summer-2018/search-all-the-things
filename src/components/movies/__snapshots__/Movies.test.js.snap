// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies test Movies renders as designed 1`] = `
<ul
  className="movies"
>
  <Movie
    key="0"
    movie={Object {}}
  />
  <Movie
    key="1"
    movie={Object {}}
  />
  <Movie
    key="2"
    movie={Object {}}
  />
</ul>
`;
